api/v1alpha1/nonadmin_types.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+// NonAdminPhase is a simple one high-level summary of the lifecycle of a NonAdminBackup or NonAdminRestore.
+// +kubebuilder:validation:Enum=New;BackingOff;Created;Deleting
+type NonAdminPhase string
+
+const (
+	// NonAdminPhaseNew - NonAdmin object was accepted by the OpenShift cluster, but it has not yet been processed by the NonAdminController
+	NonAdminPhaseNew NonAdminPhase = "New"
+	// NonAdminPhaseBackingOff - Velero object was not created due to NonAdmin object error (configuration or similar)
+	NonAdminPhaseBackingOff NonAdminPhase = "BackingOff"
+	// NonAdminPhaseCreated - Velero object was created. The Phase will not have additional information about it.
+	NonAdminPhaseCreated NonAdminPhase = "Created"
+	// NonAdminPhaseDeleting - Velero object is pending deletion. The Phase will not have additional information about it.
+	NonAdminPhaseDeleting NonAdminPhase = "Deleting"
+)
+
+// NonAdminCondition are used for more detailed information supporing NonAdminBackupPhase state.
+// +kubebuilder:validation:Enum=Accepted;Queued;Deleting
+type NonAdminCondition string
+
+// Predefined conditions for NonAdminController objects.
+// One NonAdminController object may have multiple conditions.
+// It is more granular knowledge of the NonAdminController object and represents the
+// array of the conditions through which the NonAdminController has or has not passed
+const (
+	NonAdminConditionAccepted NonAdminCondition = "Accepted"
+	NonAdminConditionQueued   NonAdminCondition = "Queued"
+	NonAdminConditionDeleting NonAdminCondition = "Deleting"
+)
+
+// QueueInfo holds the queue position for a specific operation.
+type QueueInfo struct {
+	// estimatedQueuePosition is the number of backups ahead in the queue (0 if not queued)
```

Comment:
this is going to be also used by restore? need to change comment

Reply:
@mateusoliveira43 you are right, updated to be "operations" as you suggested in other PR, I've moved the description of the actual backup to the backup part of it, similarly restore will have it:

https://github.com/migtools/oadp-non-admin/pull/130/files#diff-1fc992498fb761a2b0d175e637f5f945c2eb932ffacb0713d8724f4c14054c31R86-R89

Reply:
Ah forgot to add the change to commit. thx.